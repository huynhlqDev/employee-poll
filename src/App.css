* {
  box-sizing: border-box;
}

.App {
  display: flex;
  flex-direction: column;
  text-align: center;
}

.App-logo {
  height: 40vmin;
  pointer-events: none;
}

@media (prefers-reduced-motion: no-preference) {
  .App-logo {
    animation: App-logo-spin infinite 20s linear;
  }
}

.App-header {
  background-color: #dadfeb;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: rgb(0, 0, 0);
  max-height: 800px;
  min-height: 50px;
  margin: 0;
}
/* 
.App-link {
  color: #61dafb;
} */

@keyframes App-logo-spin {
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
}

.form-control {
  height: 40px;
  width: 400px;
  margin: 12px;
  padding: 8px;
  border-radius: 8px;
  font-size: medium;
}

.login-form-btn {
  background-color: #c140b4;
    color: #fff;
    padding: 10px 20px;
    border: none;
    border-radius: 5px;
    cursor: pointer;
    font-size: 16px;
    transition: background-color 0.3s; 
}

.login-form-btn:active {
  background-color: #cf9bc9;
}

.login-form-btn:disabled {
  background-color: #d8d3d3;
  cursor: not-allowed;
}

/* Header.css */
.header {
  position: relative;
  display: flex;
  justify-content: space-between;
  align-items: center;
  padding: 4px;
  background-color: #ffffff;
  box-shadow: 0 2px 5px rgba(0, 0, 0, 0.1);
}

.header-left {
  display: flex;
  justify-content: flex-start;
  align-items: center;
  padding-left: 20px;
}

.header-right {
  padding-right: 20px;
}

.logo-app img {
  height: 50px;
}

.header-right img {
  height: 30px;
  cursor: pointer;
}

.user-dropdown {
  position: absolute;
  top: 100%; /* Đặt dropdown ở dưới header */
  right: 0; /* Đặt dropdown ở bên phải */
  margin-top: 10px; /* Khoảng cách giữa dropdown và header */
}

.nav ul {
  list-style: none;
  margin: 0;
  padding: 0;
}

.nav ul li {
  display: inline-block;
  margin-left: 20px;
}

.nav ul li:first-child {
  margin-left: 20;
}

.nav ul li a {
  text-decoration: none;
  color: #333;
  font-weight: bold;
}

.nav ul li a:hover {
  color: #9a3397;
}


.login {
  display: flex;
  flex-direction: column;
  align-items: center;
}

.login-form {
  display: flex;
  flex-direction: column;
  align-items: stretch;
}

.login-logo {
  height: 280px;
  width: 280px;
  padding: 20px
}
