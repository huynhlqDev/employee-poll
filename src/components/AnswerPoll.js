import React, { useEffect, useState } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { useParams } from 'react-router-dom';
import { answerPoll } from '../actions/pollActions';
import PageNotFound from './PageNotFound';

const AnswerPoll = () => {
    const dispatch = useDispatch();
    const isLoading = useSelector(state => state.loading.isLoading);
    const user = useSelector(state => state.auth.user);
    const polls = useSelector(state => state.poll.polls);

    const { id } = useParams();
    const [voted, setVoted] = useState(false);
    const [userVote, setUserVote] = useState(null);
    const [currentPoll, setCurrentPoll] = useState(null);
    const [optionPercentage, setOptionPercentage] = useState(null);

    useEffect(() => {

        const checkIsVoted = () => {
            if (currentPoll) {

                const votes1 = currentPoll.optionOne?.votes
                const votes2 = currentPoll.optionTwo?.votes

                for (let i = 0; i < votes1.length; i++) {
                    if (votes1[i] === user.id) {
                        setUserVote("option one")
                        setVoted(true)
                        break;
                    }
                }
                for (let i = 0; i < votes2.length; i++) {
                    if (votes2[i] === user.id) {
                        setUserVote("option two")
                        setVoted(true)
                        break;
                    }
                }
            }
        };

        const calculatePercentage = () => {
            if (currentPoll) {
                const optionOneVoteCount = currentPoll.optionOne?.votes?.length;
                const optionTwoVoteCount = currentPoll.optionTwo?.votes?.length;
                if (optionOneVoteCount === 0 && optionTwoVoteCount === 0) {
                    return ({ percentageOptionOne: 50, percentageOptionTwo: 50 });
                }
                const percentageOptionOne = ((optionOneVoteCount / (optionOneVoteCount + optionTwoVoteCount)) * 100).toFixed(2);
                const percentageOptionTwo = ((optionTwoVoteCount / (optionOneVoteCount + optionTwoVoteCount)) * 100).toFixed(2);
                return ({ percentageOptionOne, percentageOptionTwo });
            } else {
                return null
            };
        };

        checkIsVoted();
        setOptionPercentage(calculatePercentage());

    }, [currentPoll, user, id]);

    useEffect(() => {
        setCurrentPoll(polls.find(user => user.id === id));
    }, [polls, user, id]);

    const handleAnswerPoll = (answer) => {
        if (!isLoading) {
            dispatch(answerPoll(user, currentPoll.id, answer));
        }
    };

    const paragraphStyleOptionOne = {
        width: optionPercentage?.percentageOptionOne + "%",
    };

    const paragraphStyleOptionTwo = {
        width: optionPercentage?.percentageOptionTwo + "%",
    };

    return (currentPoll ?
        <div className='create-poll-body'>
            <div className='polls-section'>
                <h3 className='polls-section-title'>Poll by {currentPoll?.author}</h3>
                {voted ?
                    <div className='poll-result'>
                        <div
                            className='option-one-area'
                            style={paragraphStyleOptionOne}
                        >{optionPercentage?.percentageOptionOne}%</div>
                        <div
                            className='option-two-area'
                            style={paragraphStyleOptionTwo}
                        >{optionPercentage?.percentageOptionTwo}%</div>
                    </div>
                    :
                    <img className='create-poll-logo'
                        src={user.avatarURL || "../create_poll.png"}
                        alt={"avatar"}
                    />
                }
                <p className='sub-title'>{voted ? `You have chosen: ${userVote}`: "Would You Rather"}</p>
                <div className='answer-poll'>
                    <div className='answer-poll-option'>
                        <p className='question-title'>{currentPoll?.optionOne?.text}</p>
                        <button
                            disabled={voted}
                            onClick={() => { handleAnswerPoll('optionOne') }}
                            className='vote-btn option-one-btn'
                        >{voted ? `votes: ${currentPoll?.optionOne?.votes?.length}` : "Click"}</button>
                    </div>
                    <div className='answer-poll-option'>
                        <p className='question-title'>{currentPoll?.optionTwo?.text}</p>
                        <button
                            disabled={voted}
                            onClick={() => { handleAnswerPoll('optionTwo') }}
                            className='vote-btn option-two-btn'
                        >{voted ? `votes: ${currentPoll?.optionTwo?.votes?.length}` : "Click"}</button>
                    </div>
                </div>
            </div>
        </div>
        :
        <PageNotFound />
    );
};

export default AnswerPoll;
