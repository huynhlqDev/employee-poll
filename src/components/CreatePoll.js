
import { useDispatch, useSelector } from 'react-redux';
import React, { useEffect, useState } from 'react';
import { savePoll, clearState } from '../actions/createPollActions';
import { Navigate } from 'react-router-dom';
import IndicatorLoading from './IndicatorLoading';


const CreatePoll = () => {
    const dispatch = useDispatch();
    const userId = useSelector(state => state.user.user.id);
    const createPoll = useSelector(state => state.createPoll);

    const [isDisabledSubmit, setIsDisabledSubmit] = useState(true)
    const [questionInfo, setQuestionInfo] = useState({ optionOneText: "", optionTwoText: "" });
    const [finish, setFinish] = useState(false);

    useEffect(() => {

        if (createPoll.result) {
            window.alert("Create success!")
            setFinish(true);
            const clearData = () => {
                dispatch(clearState())
            }
            return clearData
        } else if (createPoll.error) {
            console.log("create failed, error: ", createPoll.error);
        }

    }, [createPoll, dispatch])

    useEffect(() => {
        setIsDisabledSubmit(
            questionInfo.optionOneText === "" || questionInfo.optionTwoText === ""
        )
    }, [questionInfo])

    const handleOptionOneChange = (e) => {
        setQuestionInfo({
            ...questionInfo,
            optionOneText: e.target.value
        });
    };

    const handleOptionTwoChange = (e) => {
        setQuestionInfo({
            ...questionInfo,
            optionTwoText: e.target.value
        });
    };

    const handleCreatePoll = () => {
        dispatch(savePoll(userId, questionInfo.optionOneText, questionInfo.optionTwoText));
    };
    
    if (finish) {
        return <Navigate to="/poll-list" /> ;
    }

    return ( createPoll.loading ? <IndicatorLoading /> :
        <div className='create-poll-body'>
            <div className='polls-section '>
                <h3 className='polls-section-title'>Would You Rather</h3>
                <p className='sub-title'>Create Your Own Poll</p>
                <img className='create-poll-logo'
                    src="./create_poll.png"
                    alt={"logo"}
                />
                <div className='create-form'>
                    <div className="" id="optionOne">
                        <p className='sub-title'>First Option</p>
                        <input
                            type="text"
                            placeholder="Option one"
                            onChange={handleOptionOneChange}
                            className="create-form-control"
                        />
                    </div>
                    <div className="" id="optionTwo">
                        <p className='sub-title'>Second Option</p>
                        <input
                            type="text"
                            placeholder="Option two"
                            onChange={handleOptionTwoChange}
                            className="create-form-control"
                        />
                    </div>
                    <button disabled={isDisabledSubmit} onClick={handleCreatePoll} className='create-form-btn'>Create</button>
                </div>
            </div>
        </div>
    );
};

export default CreatePoll;
