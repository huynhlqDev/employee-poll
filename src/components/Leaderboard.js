import React, { useEffect, useState } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { getUsers } from '../actions/userActions';

const Leaderboard = () => {
    const dispatch = useDispatch();
    const isLoading = useSelector(state => state.loading.isLoading);
    const userResponse = useSelector(state => state.users.users);

    const [leadersboard, setLeadersboard] = useState([]);

    useEffect(() => {
        if (userResponse) {
            setLeadersboard(convertUsersToLeaders(userResponse));
        }
    }, [userResponse]);

    useEffect(() => {
        dispatch(getUsers())
    }, [dispatch]);

    const convertUsersToLeaders = (userResponse) => {
        return userResponse && userResponse
            .map(user => ({
                id: user.id,
                name: user.name,
                avatar: user.avatarURL,
                answersCount: Object.keys(user.answers).length,
                questionsCount: user.questions.length
            }))
            .sort((a, b) => {
                return b.answersCount - a.answersCount;
            })
    };


    return (!isLoading &&
        <div className='polls-section'>
            <h3 className='polls-section-title'>Leaderboard</h3>
            <table>
                <thead>
                    <tr>
                        <th>Users</th>
                        <th>Answered</th>
                        <th>Created</th>
                    </tr>
                </thead>
                <tbody>
                    {leadersboard.map(leader => (
                        <tr key={leader.id}>
                            <td>
                                <div className='leaderboard-card'>
                                    <img className='leaderboard-card-avatar'
                                        src={leader.avatar || "./auth-user.png"}
                                        alt={"avatar"}
                                    />
                                    <div style={{display: "flex", flexDirection: "column", marginLeft: "20px"}}>
                                        <p className='leaderboard-card-title'>{leader.name}</p>
                                        <p className='sub-title'>{leader.id}</p>
                                    </div>
                                </div>
                            </td>
                            <td>{leader.answersCount}</td>
                            <td>{leader.questionsCount}</td>
                        </tr>
                    ))}
                </tbody>
            </table>
        </div>
    );
};

export default Leaderboard;
