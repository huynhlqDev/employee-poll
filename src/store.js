// import { createStore, applyMiddleware } from 'redux';
// import rootReducer from './reducers/rootReducer';
// import { thunk } from 'redux-thunk';

// const store = createStore(rootReducer, applyMiddleware(thunk));

// export default store;

import { configureStore } from "@reduxjs/toolkit";
import userReducer from "./reduxSlice/userSlice";
import usersReducer from "./reduxSlice/usersSlice";
import pollsReducer from "./reduxSlice/pollsSlice";
import createPollReducer from "./reduxSlice/createPollSlice";
import updatePollReducer from "./reduxSlice/updatePollSlice";

export default configureStore({
    reducer: {
        user: userReducer,
        users: usersReducer,
        polls: pollsReducer,
        createPoll: createPollReducer,
        updatePoll: updatePollReducer
    }
})
