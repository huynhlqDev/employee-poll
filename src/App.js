import './App.css';
import { BrowserRouter as Router, Route, Routes, Navigate } from 'react-router-dom';
import { useSelector } from 'react-redux';
import Login from './components/Login';
import PollList from './components/PollList';
import CreatePoll from './components/CreatePoll';
import AnswerPoll from './components/AnswerPoll';
import Leaderboard from './components/Leaderboard';
import Header from './components/Header';
import IndicatorLoading from './components/IndicatorLoading';
import PageNotFound from './components/PageNotFound';
import { _getRedirecPath, _saveRedirecPath } from './data/existingUsers';

function App() {

  const isLoggedIn = useSelector(state => state.auth.isLoggedIn);

  const handleBeforeLogin = () => {
    const currentPath = window.location.pathname;
    _saveRedirecPath(currentPath);
    if (currentPath.includes("/questions")) {
      _saveRedirecPath(currentPath);
    }
    return "/login";
  }

  const handleAfterLogin = () => {
    const redirecPath = _getRedirecPath();
    if (redirecPath !== "") {
      _saveRedirecPath("")
      return redirecPath;
    } else {
      return "/poll-list";
    }
  }

  return (
    <Router>
      <Header />
      <div>
        <IndicatorLoading />
        <Routes>
          <Route path="/" exact element={isLoggedIn ? <Navigate to="/poll-list" /> : <Login />}></Route>
          <Route path="/login" element={isLoggedIn ? <Navigate to={handleAfterLogin()} /> : <Login />}></Route>
          <Route path="/poll-list" element={isLoggedIn ? <PollList /> : <Navigate to="/login" />}></Route>
          <Route path="/add" element={isLoggedIn ? <CreatePoll /> : <Navigate to="/login" />}></Route>
          <Route path="/questions/:id" element={isLoggedIn ? <AnswerPoll /> : <Navigate to={handleBeforeLogin()} />}></Route>
          <Route path="/leaderboard" element={isLoggedIn ? <Leaderboard /> : <Navigate to='/login' />}></Route>
          <Route path="/PageNotFound" element={<PageNotFound />} />
          <Route path="*" element={<PageNotFound />} />
        </Routes>
      </div>
    </Router>
  );
}

export default App;
