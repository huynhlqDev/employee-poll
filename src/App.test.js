import { render, screen, fireEvent } from '@testing-library/react';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';
import { thunk } from 'redux-thunk';

import App from './App';
import store from './store';
import Login from './components/Login';
import PollList from './components/PollList';
import Leaderboard from './components/Leaderboard';
import CreatePoll from './components/CreatePoll';
import AnswerPoll from './components/AnswerPoll';

// Test DOM component
describe('Test DOM component', () => {
  it("DOM test which uses the fireEvent function", () => {
    var app = render(
      <Provider store={store}>
        <App />
      </Provider>
    );
  
    // verify snapshot
    expect(app).toMatchSnapshot();
  
    // verify DOM component
    const loginTitle = screen.getByText(/Employee Polls/);
    const submitButton = screen.getByText(/Sign in/);
    const usernameInput = screen.getByPlaceholderText(/Username/);
    const passwordInput = screen.getByPlaceholderText(/Password/);
  
    expect(loginTitle).toBeInTheDocument();
    expect(submitButton).toBeInTheDocument();
    expect(usernameInput).toBeInTheDocument();
    expect(passwordInput).toBeInTheDocument();
  
    // DOM fire event
    fireEvent.change(usernameInput, { target: { value: 'zoshikanlu' } });
    fireEvent.change(passwordInput, { target: { value: 'pass246' } });
    fireEvent.click(submitButton);
    
    // Confirm hiding the login page when submitted successfully
    expect(loginTitle).not.toBeInTheDocument();
    expect(usernameInput).not.toBeInTheDocument();
    expect(passwordInput).not.toBeInTheDocument();
    expect(submitButton).not.toBeInTheDocument();
  })
  
});

// Test: component snapshot
describe("matches the snapshot when render component", () => {
  //Test: Login component
  it("Login", () => {
    var component = render(
      <Provider store={store}>
        <Login />
      </Provider>
    );

    expect(component).toMatchSnapshot();
  })

  //Test: PollList component
  it("Home", () => {
    const mockStore = configureMockStore([thunk]);
    const initialState = {
      poll: {
        polls: []
      },
      loading: {
        isLoading: false
      },
      auth: {
        user: {
          answers: {}
        }
      }
    };

    const store = mockStore(initialState);
    var component = render(
      <Provider store={store}>
        <PollList />
      </Provider>
    );

    expect(component).toMatchSnapshot();
  })


  //Test: CreatePoll component
  it("Create poll", () => {
    const mockStore = configureMockStore([thunk]);
    const initialState = {
      auth: {
        isLoggedIn: false,
        user: { id: "", password: "" }
      },
      createPoll: {
        createPoll: false
      }
    };
    const store = mockStore(initialState);
    var component = render(
      <Provider store={store}>
        <CreatePoll />
      </Provider>
    );

    expect(component).toMatchSnapshot();
  })


  //Test: AnswerPoll component
  it("Answer poll", () => {
    var component = render(
      <Provider store={store}>
        <AnswerPoll />
      </Provider>
    );

    expect(component).toMatchSnapshot();
  })


  //Test: Leadersboard component
  it("leadersboard", () => {
    var component = render(
      <Provider store={store}>
        <Leaderboard />
      </Provider>
    );

    expect(component).toMatchSnapshot();
  })

})