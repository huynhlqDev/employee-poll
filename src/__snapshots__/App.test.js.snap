// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test DOM component DOM test which uses the fireEvent function 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <header
        class="header"
      >
        <div
          class="header-left"
        >
          <div
            class="logo-app"
          >
            <a
              href="/"
            >
              <img
                alt="Logo"
                src="/polls-logo.png"
              />
            </a>
          </div>
        </div>
        <div
          class="header-right"
        >
          <div
            class="login-box"
          >
            <img
              alt="Logo"
              src="/login-logo.png"
            />
          </div>
        </div>
      </header>
      <div>
        <div
          class="login"
        >
          <h1>
            Employee Polls
          </h1>
          <img
            alt="logo"
            class="login-logo"
            src="./login-logo-2.png"
          />
          <div
            class="login-form"
          >
            <div
              class="login-input"
            >
              <input
                class="form-control"
                id="username"
                placeholder="Username"
                type="text"
              />
            </div>
            <div
              class="login-input"
            >
              <input
                class="form-control"
                id="password"
                placeholder="Password"
                type="password"
              />
            </div>
            <button
              class="login-form-btn"
              disabled=""
            >
              Submit
            </button>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <header
      class="header"
    >
      <div
        class="header-left"
      >
        <div
          class="logo-app"
        >
          <a
            href="/"
          >
            <img
              alt="Logo"
              src="/polls-logo.png"
            />
          </a>
        </div>
      </div>
      <div
        class="header-right"
      >
        <div
          class="login-box"
        >
          <img
            alt="Logo"
            src="/login-logo.png"
          />
        </div>
      </div>
    </header>
    <div>
      <div
        class="login"
      >
        <h1>
          Employee Polls
        </h1>
        <img
          alt="logo"
          class="login-logo"
          src="./login-logo-2.png"
        />
        <div
          class="login-form"
        >
          <div
            class="login-input"
          >
            <input
              class="form-control"
              id="username"
              placeholder="Username"
              type="text"
            />
          </div>
          <div
            class="login-input"
          >
            <input
              class="form-control"
              id="password"
              placeholder="Password"
              type="password"
            />
          </div>
          <button
            class="login-form-btn"
            disabled=""
          >
            Submit
          </button>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`matches the snapshot when render component Answer poll 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div />
  </body>,
  "container": <div />,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`matches the snapshot when render component Create poll 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="login"
      >
        <h1>
          Would You Rather
        </h1>
        <p>
          Create Your Own Poll
        </p>
        <img
          alt="logo"
          class="login-logo"
          src="./login-logo-2.png"
        />
        <div
          class="login-form"
        >
          <div
            class="login-input"
            id="optionOne"
          >
            <p>
              First Option
            </p>
            <input
              class="form-control"
              placeholder="Option one"
              type="text"
            />
          </div>
          <div
            class="login-input"
            id="optionTwo"
          >
            <p>
              Second Option
            </p>
            <input
              class="form-control"
              placeholder="Option two"
              type="text"
            />
          </div>
          <button
            class="login-form-btn"
            disabled=""
          >
            Submit
          </button>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="login"
    >
      <h1>
        Would You Rather
      </h1>
      <p>
        Create Your Own Poll
      </p>
      <img
        alt="logo"
        class="login-logo"
        src="./login-logo-2.png"
      />
      <div
        class="login-form"
      >
        <div
          class="login-input"
          id="optionOne"
        >
          <p>
            First Option
          </p>
          <input
            class="form-control"
            placeholder="Option one"
            type="text"
          />
        </div>
        <div
          class="login-input"
          id="optionTwo"
        >
          <p>
            Second Option
          </p>
          <input
            class="form-control"
            placeholder="Option two"
            type="text"
          />
        </div>
        <button
          class="login-form-btn"
          disabled=""
        >
          Submit
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`matches the snapshot when render component Home 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="polls"
      >
        <div
          class="polls-section"
        >
          <h3
            class="polls-section-title"
          >
            New Questions
          </h3>
          <div
            class="polls-section-body"
          />
        </div>
        <div
          class="polls-section"
        >
          <h3
            class="polls-section-title"
          >
            Done
          </h3>
          <div
            class="polls-section-body"
          />
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="polls"
    >
      <div
        class="polls-section"
      >
        <h3
          class="polls-section-title"
        >
          New Questions
        </h3>
        <div
          class="polls-section-body"
        />
      </div>
      <div
        class="polls-section"
      >
        <h3
          class="polls-section-title"
        >
          Done
        </h3>
        <div
          class="polls-section-body"
        />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`matches the snapshot when render component Login 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div />
  </body>,
  "container": <div />,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`matches the snapshot when render component leadersboard 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div />
  </body>,
  "container": <div />,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
